src/backend/metal/RenderPipelineMTL.mm

```diff
@@ -193,4 +197,58 @@ MTLVertexFormat toMTLVertexFormat(VertexFormat vertexFormat)
     colorAttachmentDescriptor.destinationAlphaBlendFactor = _blendDescriptorMTL.destinationAlphaBlendFactor;
 }
 
+uint32_t RenderPipelineMTL::createHash(const RenderPipelineDescriptor& descriptor)
+{
+    std::string info = "";
```

Comment:
I think these codes is better in `RenderPipelineDescriptor`.

Reply:
Since RenderPipelineDescriptor was defined in pure cpp file, but i need to do some static_cast to xxxMTL object. Thus createHash method was defined in RenderPipelineMTL just for simplify.